Providers/IntercomProvider.m

```diff
@@ -0,0 +1,133 @@
+//
+//  IntercomProvider.m
+
+#import "IntercomProvider.h"
+#import "Intercom.h"
+
+static NSString *const kIntercomSuperPropertiesKey = @"superProperties";
+
+@interface IntercomProvider ()
+
+@property(nonatomic, strong) NSDictionary *superProperties;
+
+@end
+
+
+@implementation IntercomProvider
+
+#ifdef AR_INTERCOM_EXISTS
```

Comment:
The ARAnalytics library is doing it as part of the cocoapod spec. It is the same way all other providers behave.